### PR TITLE
Allow fluid variant colors to depend on the position

### LINKED ISSUE #1582: Combine `FluidVariantRenderHandler` and `FluidRenderHandler`
@shedaniel noted that world and blockpos parameters are missing from `FluidVariantRenderHandler`, whereas a `@Nullable NbtCompound` parameter is missing from `FluidRenderHandler`. And we can get rid of the `FluidState` parameter as it can be queried from the world if needed with close to 0 overhead.

So we should combine both, probably by expanding `FluidRenderHandler`. We can then remove `FluidVariantRenderHandler` in a further iteration of the transfer API.

### PR DESCRIPTION
**Pull request by Technici4n**
Closes #1582.

Also adds a helper `FluidVariantRendering.getTooltip(FluidVariant)`.

### CODE REVIEW

File: fabric-transfer-api-v1/src/main/java/net/fabricmc/fabric/api/transfer/v1/client/fluid/FluidVariantRenderHandler.java [hunk 1]
```diff
@@ -76,14 +78,29 @@ default Sprite getSprite(FluidVariant fluidVariant) {
...
+	 * If they are provided, this method may return a color that depends on the location.
+	 * For example, water returns the biome-dependent color if the context parameters are specified, or its default color if one of them is null.
+	 */
+	default int getColor(FluidVariant fluidVariant, @Nullable BlockRenderView view, @Nullable BlockPos pos) {
 		// Use the fluid render handler by default.
 		FluidRenderHandler fluidRenderHandler = FluidRenderHandlerRegistry.INSTANCE.get(fluidVariant.getFluid());
 
 		if (fluidRenderHandler != null) {
-			return fluidRenderHandler.getFluidColor(null, null, fluidVariant.getFluid().getDefaultState());
+			return fluidRenderHandler.getFluidColor(view, pos, fluidVariant.getFluid().getDefaultState()) | 255 << 24;
```
Comment (Juuxel) [inline fabric-transfer-api-v1/src/main/java/net/fabricmc/fabric/api/transfer/v1/client/fluid/FluidVariantRenderHandler.java:103]: Hmm, should there be an option to pass in the fluid state to the original handler as well?

Comment (Technici4n) [reply fabric-transfer-api-v1/src/main/java/net/fabricmc/fabric/api/transfer/v1/client/fluid/FluidVariantRenderHandler.java:103]: I don't think a `FluidVariant` can be mapped to a `FluidState` easily. If the fluid state depends on the NBT of the variant, then I'd suggest that a custom `FluidVariantRenderHandler` be registered by the fluid.

**Review by Juuxel — APPROVED** [review]

File: fabric-transfer-api-v1/src/testmod/java/net/fabricmc/fabric/test/transfer/fluid/client/FluidVariantRenderTest.java [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+	private static void drawFluidInGui(MatrixStack ms, FluidVariant fluid, World world, BlockPos pos, int i, int j) {
+		RenderSystem.setShaderTexture(0, SpriteAtlasTexture.BLOCK_ATLAS_TEXTURE);
+		Sprite sprite = FluidVariantRendering.getSprite(fluid);
+		int color = FluidVariantRendering.getColor(fluid, world, pos);
+
+		if (sprite == null) return;
+
+		float r = ((color >> 16) & 255) / 256f;
+		float g = ((color >> 8) & 255) / 256f;
+		float b = (color & 255) / 256f;
```
Comment (liach) [inline fabric-transfer-api-v1/src/testmod/java/net/fabricmc/fabric/test/transfer/fluid/client/FluidVariantRenderTest.java]: Is it really 256f than 255f?

Comment (Technici4n) [reply fabric-transfer-api-v1/src/testmod/java/net/fabricmc/fabric/test/transfer/fluid/client/FluidVariantRenderTest.java]: Oops :smile:

Comment (liach) [reply fabric-transfer-api-v1/src/testmod/java/net/fabricmc/fabric/test/transfer/fluid/client/FluidVariantRenderTest.java]: It's fine, it's just a test :smile:

**Review by liach — APPROVED** [review]

**Review by modmuss50 — APPROVED** [review]

**Pull request merged**